src/WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -220,8 +219,36 @@ public static void InitializeComWrappers(ComWrappers wrappers = null)
                 return obj => obj;
             }
 
-            var constructor = implementationType.GetConstructor(BindingFlags.NonPublic | BindingFlags.CreateInstance | BindingFlags.Instance, null, new[] { typeof(IObjectReference) }, null);
-            return (IInspectable obj) => constructor.Invoke(new[] { obj.ObjRef });
+            var attribute = implementationType.GetCustomAttribute<WinRTImplementationTypeRcwFactoryAttribute>();

```

Comment:
inherit should probably be false here to get the tests to pass.

Reply:
Fixed in https://github.com/microsoft/CsWinRT/pull/1505/commits/cfc12166fb462d15769d93167974d367c6a5e30d 🙂